dali/operators/segmentation/random_mask_pixel.cc

```diff
@@ -0,0 +1,192 @@
+  }
+}
+
+bool RandomMaskPixelCPU::SetupImpl(std::vector<OutputDesc> &output_desc,
+                                    const workspace_t<CPUBackend> &ws) {
+  const auto &in_masks = ws.template InputRef<CPUBackend>(0);
+  int nsamples = in_masks.size();
+  auto in_masks_shape = in_masks.shape();
+  int ndim = in_masks_shape.sample_dim();
+  output_desc.resize(1);
+  output_desc[0].shape = uniform_list_shape(nsamples, {ndim});
+  output_desc[0].type = TypeTable::GetTypeInfo(DALI_INT64);
+
+  foreground_.resize(nsamples);
+  value_.clear();
+  threshold_.clear();
+
+  GetPerSampleArgument(foreground_, "foreground", ws, nsamples);
+  if (spec_.ArgumentDefined("value")) {
+    GetPerSampleArgument(value_, "value", ws, nsamples);
+  } else {
+    GetPerSampleArgument(threshold_, "threshold", ws, nsamples);
+  }
+  return true;
+}
+
+template <typename T>
+void RandomMaskPixelCPU::RunImplTyped(workspace_t<CPUBackend> &ws) {
+  const auto &in_masks = ws.template InputRef<CPUBackend>(0);
+  auto &out_pixel_pos = ws.template OutputRef<CPUBackend>(0);
+  int nsamples = in_masks.size();
+  auto in_masks_shape = in_masks.shape();
+  int ndim = in_masks_shape.sample_dim();
+  auto masks_view = view<const T>(in_masks);
+  auto pixel_pos_view = view<int64_t>(out_pixel_pos);
+  auto& thread_pool = ws.GetThreadPool();
+
+  if (rng_.empty()) {
+    for (int i = 0; i < thread_pool.size(); i++) {
+      rng_.emplace_back(seed_ + i);
+    }
+  }
+  assert(rng_.size() == static_cast<size_t>(thread_pool.size()));
+  rle_.resize(thread_pool.size());
+
+  for (int sample_idx = 0; sample_idx < nsamples; sample_idx++) {
+    thread_pool.AddWork(
+      [&, sample_idx](int thread_id) {
+        auto &rng = rng_[thread_id];
+        auto mask = masks_view[sample_idx];
+        auto pixel_pos = pixel_pos_view[sample_idx];
+        const auto &mask_sh = mask.shape;
+        if (foreground_[sample_idx]) {
+          int64_t flat_idx = -1;
+          auto &rle_mask = rle_[thread_id];
+          if (has_value_) {
+            T value = static_cast<T>(value_[sample_idx]);
+            // checking if the value is representable by T, otherwise we
+            // just fall back to pick a random pixel.
+            if (static_cast<int>(value) == value_[sample_idx]) {
+              rle_mask.Init(
+                  mask, [value](const T &x) { return x == value; });
+              if (rle_mask.count() > 0) {
+                auto dist = std::uniform_int_distribution<int64_t>(0, rle_mask.count() - 1);
+                flat_idx = rle_mask.find(dist(rng));
+              }
+            }
+          } else {
+            float threshold = threshold_[sample_idx];
+            rle_mask.Init(
+                mask, [threshold](const T &x) { return x > threshold; });
+            if (rle_mask.count() > 0) {
+              auto dist = std::uniform_int_distribution<int64_t>(0, rle_mask.count() - 1);
+              flat_idx = rle_mask.find(dist(rng));
+            }
+          }
```

Comment:
```suggestion
          auto &rle_mask = rle_[thread_id];
          rle_mask.clear();
          if (has_value_) {
            T value = static_cast<T>(value_[sample_idx]);
            // checking if the value is representable by T, otherwise we
            // just fall back to pick a random pixel.
            if (static_cast<int>(value) == value_[sample_idx]) {
              rle_mask.Init(
                  mask, [value](const T &x) { return x == value; });
            }
          } else {
            float threshold = threshold_[sample_idx];
            rle_mask.Init(
                mask, [threshold](const T &x) { return x > threshold; });
          }
          if (rle_mask.count() > 0) {
            auto dist = std::uniform_int_distribution<int64_t>(0, rle_mask.count() - 1);
            flat_idx = rle_mask.find(dist(rng));
          }
```